odysseus/odysseus_tree/board/tpu/plugin_user_property_timestamp.c

```diff
@@ -0,0 +1,48 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/time.h>
+
+
+#include "nanomq/plugin.h"
+
+/*
+ * How to compile:
+ * gcc -I../ -fPIC -shared plugin_user_property.c -o plugin_user_property.so
+ */
+
+
+int cb(void *data)
```

Comment:
why is this void *

Reply:
Idk it is in upstream.  The source function: https://github.com/nanomq/nanomq/blob/d271182eb11f8a9393f051f359e750463a090d63/nanomq/plugin/plugin.c#L19